src/components/sidenav/sidenav.scss

```diff
@@ -0,0 +1,151 @@
+@import "default-theme";
+@import "mixins";
+@import "variables";
+@import "shadows";
+
+
+$md-sidenav-background-color: md-color($md-background, 100) !default;
+$md-sidenav-push-background-color:  md-color($md-background, 100) !default;
+
+
+/**
+ * Mixin to help with defining LTR/RTL `transform: translateX()` values.
+ * @param $open The translation value when the sidenav is opened.
+ * @param $close The translation value when the sidenav is closed.
+ */
+@mixin md-sidenav-transition($open, $close) {
+  transform: translateX($close);
+
+  &.md-sidenav-closed {
+    /**
+     * We use `visibility: hidden | visible` because `display: none` will not animate any
+     * transitions, while visibility will interpolate transitions properly.
+     * see https://developer.mozilla.org/en-US/docs/Web/CSS/visibility, the Interpolation
+     * section.
+     */
```

Comment:
We should set a consistent pattern for block-style (`/** */`) vs. inline-style (`//`) comments. I propose using block-style when providing a description of a class / mixin / function (similar to JsDoc), and to use an inline--style comment when you're just explaining an approach.